File: src/UE4_Packages/FPerPlatformFloat.java
/**
 * 
 */
package UE4_Packages;

/**
 * @author FunGames
 *
 */
public class FPerPlatformFloat {
	private boolean cooked;
	private float value;
	
	public FPerPlatformFloat(FArchive Ar) throws ReadException {
		cooked = Ar.readBooleanFromUInt8();
		value = Ar.readFloat32();
	}
	public boolean getCooked() {
		return cooked;
	}
	public float getValue() {
		return value;
	}
}

File: src/main/kotlin/me/fungames/jfortniteparse/ue4/assets/objects/FPerPlatformFloat.kt
package me.fungames.jfortniteparse.ue4.assets.objects

import me.fungames.jfortniteparse.ue4.UClass
import me.fungames.jfortniteparse.ue4.reader.FArchive
import me.fungames.jfortniteparse.ue4.writer.FArchiveWriter

@ExperimentalUnsignedTypes
class FPerPlatformFloat : UClass {
    var cooked : Boolean
    var value: Float

    constructor(Ar: FArchive) {
        super.init(Ar)
        cooked = Ar.readFlag()
        value = Ar.readFloat32()
        super.complete(Ar)
    }

    fun serialize(Ar: FArchiveWriter) {
        super.initWrite(Ar)
        Ar.writeFlag(cooked)
        Ar.writeFloat32(value)
        super.completeWrite(Ar)
    }

    constructor(cooked : Boolean, value : Float) {
        this.cooked = cooked
        this.value = value
    }
}